pkg/network/ebpf/c/protocols/kafka/kafka-parsing.h

```diff
@@ -507,9 +604,16 @@ static __always_inline enum parse_result kafka_continue_parse_response_loop(kafk
         case KAFKA_FETCH_RESPONSE_PARTITION_END:
             response->partitions_count--;
             if (response->partitions_count == 0) {
-                extra_debug("enqueue, records_count %d",  response->transaction.records_count);
+                extra_debug("enqueue final partition, records_count %d",  response->transaction.records_count);
                 kafka_batch_enqueue_wrapper(kafka, tup, &response->transaction);
                 return RET_DONE;
+            } else {
+                extra_debug("enqueue partition, #partitions left %d, records_count %d",
+                    response->partitions_count,
+                    response->transaction.records_count);
+                kafka_batch_enqueue_wrapper(kafka, tup, &response->transaction);
```

Comment:
Won't queuing after every partition cause a large increase in events, and possible even cause the batching to be full if there are many partitions? For example, the "many partitions" test in the fetch v12 patchset which may fail with this change.

Perhaps we could at the very least optimize this for the common case of no error by only queuing the event if the error code of the current partition is different from the error codes of the previous partitions? So the enqueue would be done in `KAFKA_FETCH_RESPONSE_PARTITION_ERROR_CODE_START` instead (and at the end).

Reply:
The optimization is a great idea, I'll implement it